main.py

```diff
@@ -278,4 +301,87 @@ def default_out_parser(metadata_dict, filename, metadata_category):
   fileobj.close()
 
 
+def get_command_line_args():
+  """
+  <Purpose>
+    Uses the argparse module to parse, process and return the command line 
+    arguments.
+
+  <Arguments>
+    None.
+
+  <Exceptions>
+    TBD.
+
+  <Return>
+    An object representing the command line arguments.
+  """
+
+  parser = argparse.ArgumentParser(prog='main.py', description='Captures the given build/test command\'s I/O and relevant system details.')
+  parser.add_argument('--version', action='version', version=TOTO_TOOL_VERSION)
+  parser.add_argument('--input', metavar='FILEPATH', help='the path to the desired input file to be routed through stdin')
+  parser.add_argument('--policy', metavar='FILENAME', help='the path to the desired file specifying build/test policy')
+  parser.add_argument('command', metavar='COMMAND', type=str, help='the bash command to execute the build or test')
+
+  return parser.parse_args()
+
+
+def process_policy_file(metadata_dict, policy_filepath):
```

Comment:
Personally, I think we'd want to throw the exception and let it fail, meaning no metadata is generated. The way I see it, they impose these constraints on themselves, so they wouldn't want to allow the process to finish until these constraints are met. If they don't care whether the constraint is met or not and want metadata either way, then they just wouldn't set the constraint field in the first place. 

What do you think?
@ceinfo 
@Justinvalcarcel